receiver/collectdreceiver/receiver_test.go

```diff
@@ -172,6 +173,33 @@ func TestCollectDServer(t *testing.T) {
 			assertMetricsAreEqual(t, tt.wantData, mds)
 		})
 	}
+	testInvalidMethod(t, sink)
+}
+
+func testInvalidMethod(t *testing.T, sink *consumertest.MetricsSink) {
+
+	testInvlidMethodCase := testCase{
+		name:         "invalid-request-body",
+		requestBody:  `invalid-body`,
+		responseCode: 400,
+		wantData:     []pmetric.Metrics{},
+	}
+
+	t.Run(testInvlidMethodCase.name, func(t *testing.T) {
+		sink.Reset()
+		req, err := http.NewRequest(
+			"GET",
```

Comment:
Can we parameterize this method as another `testCase` field instead of adding another test case?